packages/playwright-core/src/server/recorder/recorderApp.ts

```diff
@@ -61,24 +61,26 @@ export class RecorderApp extends EventEmitter implements IRecorderApp {
   private async _init() {
     await syncLocalStorageWithSettings(this._page, 'recorder');
 
-    await this._page._setServerRequestInterceptor(route => {
-      if (!route.request().url().startsWith('https://playwright/'))
-        return false;
-
-      const uri = route.request().url().substring('https://playwright/'.length);
-      const file = require.resolve('../../vite/recorder/' + uri);
-      fs.promises.readFile(file).then(buffer => {
-        route.fulfill({
-          status: 200,
-          headers: [
-            { name: 'Content-Type', value: mime.getType(path.extname(file)) || 'application/octet-stream' }
-          ],
-          body: buffer.toString('base64'),
-          isBase64: true
-        }).catch(() => {});
+    if (!process.env.PW_HMR) {
```

Comment:
Perhaps if we enable interception when HMR is on, nothing will break? If so, we can remove the extra `if`.

Reply:
Fair point - addressed.